Source/Bogus/DataSets/Date.cs

```diff
@@ -144,31 +129,221 @@ public DateTimeOffset FutureOffset(int yearsToGoForward = 1, DateTimeOffset? ref
+            {
+               start = new DateTime(window.Start.Ticks, start.Kind);
+
+               if (start == end)
+                  return;
+            }
+
+            window = GetForwardDSTTransitionWindow(end);
+
+            if ((end >= window.Start) && (end < window.End))
+            {
+               end = new DateTime(window.End.Ticks, end.Kind);
+
+               // We had to bump the end, meaning that the end was not already a valid
+               // DateTime value (within the DST transition range), so prefer the start
+               // instead.
+               preferRangeBoundary = start;
+            }
+
+            if (start > end)
+               throw new Exception("DateTime range does not contain any real DateTime values due to daylight savings transitions");
+         }
+#endif
+      }
+
+#if !NETSTANDARD1_3
+      struct DateTimeRange
+      {
+         public DateTime Start;
+         public DateTime End;
       }
 
+      /// <summary>
+      /// Finds the window of time that doesn't exist in the local timezone due to Daylight Savings Time coming into
+      /// effect. In timezones that do not have Daylight Savings Time transitions, this function returns <see cref="null"/>.
+      /// </summary>
+      /// <param name="dateTime">
+      /// A reference <see cref="DateTime"/> value for determining the DST transition window accurately. Daylight Savings Time
+      /// rules can change over time, and the <see cref="TimeZoneInfo"/> API exposes information about which Daylight Savings
+      /// Time rules are in effect for which date ranges.
+      /// </param>
+      /// <returns>
+      /// A <see cref="DateTimeRange"/> that indicates the start &amp; end of the interval of date/time values that do not
+      /// exist in the local calendar in the interval indicated by the supplied <paramref name="dateTime"/>, or <see cref="null"/>
+      /// if no such range exists.
+      /// </returns>
+      private DateTimeRange GetForwardDSTTransitionWindow(DateTime dateTime)
+      {
+         // Based on code found at: https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.transitiontime.isfixeddaterule
+         var rule = FindEffectiveTimeZoneAdjustmentRule(dateTime);
+
+         if (rule == null)
+            return default(DateTimeRange);
+
+         var transition = rule.DaylightTransitionStart;
+
+         DateTime startTime;
+
+         if (transition.IsFixedDateRule)
+         {
+            startTime = new DateTime(
+               dateTime.Year,
+               transition.Month,
+               transition.Day,
+               transition.TimeOfDay.Hour,
+               transition.TimeOfDay.Minute,
+               transition.TimeOfDay.Second,
+               transition.TimeOfDay.Millisecond);
+         }
+         else
+         {
+            var calendar = CultureInfo.CurrentCulture.Calendar;
+
+            var startOfWeek = transition.Week * 7 - 6;
```

Comment:
Not sure if such `Calendar`s are implemented, but there are real-world calendars which do not follow a 7-day week.

Reply:
The same is relevant for a few more lines in this method.